### PR TITLE
backup feature

### PR DESCRIPTION
**Pull request by tarkah**
This feature will allow the user to backup their Addons and WTF folder through Ajour. Once the user selects a backup directory in settings, a "Backup Now" button will be available to push. Once clicked, the app will zip the Addons and WTF folders from each flavor present on the system and save that zip file under the backup folder with name `ajour_backup_YYYY-MM-DD_HH-MM-SS.zip`. We can then parse this folder on startup and determine the most recent backup made by the user and present that datetime along with the Backup Now button.

This won't manage the backups once they are created, so it won't handle pruning old backups or restoring from backups.

### CODE REVIEW

**Comment by tarkah** [comment]
@casperstorm Let me know if you agree with my outline of this PR above. My first commit handles the logic for "backing" up folders with zip format. I've implemented a test for this backup, but we will probably want to remove that before this PR goes live since it operates on the actual fs.

**Comment by casperstorm** [comment]
> Let me know if you agree with my outline of this PR above. My first commit handles the logic for "backing" up folders with zip format. I've implemented a test for this backup, but we will probably want to remove that before this PR goes live since it operates on the actual fs.

Looks good to me, @tarkah. Exactly how I thought of it as well. Looks nice and clean as well.
I agree with removing the test once we are ready to merge this.

**Comment by tarkah** [comment]
All done! I think it works pretty well. The only change I'd like to make is to have some sort of "animation" that the backup process is in progress, vs the static text I currently show. I haven't really looked into how that can be done with Iced.

Also, I expanded the Settings height to fit the backup options in the right column, and leave enough space to show 4 ignored addons within the scrollable. Hopefully we don't have too much more to squeeze into the settings pane :P

**Comment by tarkah** [comment]
Some screenshots to show the change from No Directory Selected -> Selected -> Backing Up -> Backup Done

image
image
image
image

File: src/backup.rs
Comment (casperstorm) [inline src/backup.rs:34]: I like that we find the latest saved rather than saving it to the config. Good thinking here @tarkah.

File: CHANGELOG.md
Comment (casperstorm) [inline CHANGELOG.md:25]: You remembered it 😀

File: src/gui/mod.rs
Comment (casperstorm) [inline src/gui/mod.rs:45]: 👍🏻

**Review by casperstorm — APPROVED** [review]
This is tested on both Windows and macOS.
On Windows I had no issues at all, but on macOS it seemed like there was a problem unzipping. I was using the default `Archive Utility.app` – i then tried `The Unarchiver.app` a popular third party and that worked. I also used the default `unzip` command from the terminal and that worked as well so I believe it must be (yet another) bug in either macOS or `Archive Utility.app` – I don't want to hold this feature back because of it.

Good job, @tarkah – really a nice solution you came up with.

**Comment by tarkah** [comment]
Interesting, you'd think the default app would work. I'm using the bzip2 compression method, maybe it doesn't support that? Anyway, I also tried on linux and confirmed it worked with the gnome archive manager.

**Comment by casperstorm** [comment]
> Interesting, you'd think the default app would work. I'm using the bzip2 compression method, maybe it doesn't support that? Anyway, I also tried on linux and confirmed it worked with the gnome archive manager.

https://github.com/casperstorm/ajour/pull/132

**Pull request merged**
